File: Strings/812-rotate-string/rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        int n = goal.length();
        int m = s.length();

        if(n!=m){
            return false;
        }
        
        String doubles = s + s;
        if(doubles.contains(goal))
        {
            return true;
        }
        return false;
    }
}

File: Strings/812-rotate-string/rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        l1 = len(s)
        l2 = len(goal)

        if (l1 != l2):
            return False

        new_string = s + s

        if goal in new_string:
            return True
        else:
            return False